packages/react/src/Banner/Banner.tsx

```diff
@@ -64,74 +71,96 @@ const iconForVariant: Record<BannerVariant, React.ReactNode> = {
   warning: <AlertIcon />,
 }
 
+const labels: Record<BannerVariant, string> = {
+  critical: 'Critical',
+  info: 'Information',
+  success: 'Success',
+  upsell: 'Recommendation',
+  warning: 'Warning',
+}
+
 export const Banner = React.forwardRef<HTMLElement, BannerProps>(function Banner(
-  {children, description, hideTitle, icon, onDismiss, primaryAction, secondaryAction, title, variant = 'info', ...rest},
-  ref,
+  {
+    'aria-label': label,
+    children,
+    description,
+    hideTitle,
+    icon,
+    onDismiss,
+    primaryAction,
+    secondaryAction,
+    title,
+    variant = 'info',
+    ...rest
+  },
+  forwardRef,
 ) {
-  const titleId = useId()
-  const value = useMemo(() => {
-    return {
-      titleId,
-    }
-  }, [titleId])
   const dismissible = variant !== 'critical' && onDismiss
   const hasActions = primaryAction || secondaryAction
+  const bannerRef = React.useRef<HTMLElement>(null)
+  const ref = useMergedRefs(forwardRef, bannerRef)
 
   if (__DEV__) {
-    // Note: __DEV__ will make it so that this hook is consistently called, or
-    // not called, depending on environment
+    // This hook is called consistently depending on the environment
     // eslint-disable-next-line react-hooks/rules-of-hooks
     useEffect(() => {
-      const title = document.getElementById(titleId)
-      if (!title) {
+      if (title) {
+        return
+      }
+
+      const {current: banner} = bannerRef
+      if (!banner) {
+        return
+      }
+
+      const hasTitle = banner.querySelector('[data-banner-title]')
+      if (!hasTitle) {
         throw new Error(
-          'The Banner component requires a title to be provided as the `title` prop or through `Banner.Title`',
+          'Expected a title to be provided to the <Banner> component with the `title` prop or through `<Banner.Title>` but no title was found',
```

Comment:
What is a good way to differentiate `Error` vs `console.error (or invariant)` in these type of scenarios like when to use one over the other one? 👀

Reply:
@broccolinisoup great question honestly, they kind of each have a niche and I'm not sure when it's best to use which 🤔 Would this be a good topic for a working session?

Reply:
I think it is a great idea 🔥 I had related questions in mind as well. I'll add it to the agenda!